# Topic & Frame
global_path_topic: "/move_base/GlobalPlanner/plan"
odom_topic: "/odom_gazebo"

odom_frame: "odom"
car_frame: "base_link"

# Parameters for control loop
pub_twist_cmd: true
debug_info: false
time_spent: true
max_speed: 1.5 # unit: m/s 
waypoints_dist: -1.0 # unit: m, set < 0 means computed by node
path_length: 5.0 # unit: m
goal_radius: 0.5 # unit: m
controller_freq: 10

# Parameter for MPC solver
mpc_steps: 20.0
mpc_ref_cte: 0.0
mpc_ref_vel: 0.5
mpc_ref_etheta: 0.0

mpc_w_cte: 500.0
mpc_w_etheta: 1000.0
mpc_w_vel: 1000.0
mpc_w_angvel: 1000.0
mpc_w_angvel_d: 1000.0
mpc_w_accel: 100.0
mpc_w_accel_d: 0.0

mpc_max_angvel: 1.5 
mpc_max_throttle: 1.0 # Maximal throttle accel
mpc_bound_value: 1.0e3 # Bound value for other variables